src/main/java/solrocr/OcrHighlighter.java

```diff
@@ -528,14 +528,20 @@ protected List<IterableCharSequence[]> loadOcrFieldValues(
           ocrVals[fieldIdx] = IterableCharSequence.fromString(fieldValue);
           continue;
         }
-        SourcePointer sourcePointer = SourcePointer.parse(fieldValue);
+        SourcePointer sourcePointer = null;
+        try {
+          sourcePointer = SourcePointer.parse(fieldValue);
+        } catch (RuntimeException e) {
+          log.error("Could not parse OCR pointer for document {}: {}", docId, fieldValue, e);
+        }
         if (sourcePointer == null) {
           // None of the files in the pointer exist or were readable, log should have warnings
           ocrVals[fieldIdx] = null;
           continue;
         }
         // If preloading is enabled, start warming the cache for the pointer
-        PageCacheWarmer.getInstance().ifPresent(w -> w.preload(sourcePointer));
+        final SourcePointer finalPtr = sourcePointer;
+        PageCacheWarmer.getInstance().ifPresent(w -> w.preload(finalPtr));
```

Comment:
Variables used in lambdas have to be final or "effectively" final. Since we separated the declaration and assignment for the variable in this PR, it's no longer effectively final and we have to create a final alias to be able to use it in the lambda

Reply:
Thanks for the explanation :)